HVZ.Web/Shared/Ui Helpers/ThemeToggle.razor

```diff
@@ -0,0 +1,38 @@
+﻿@inject IJSRuntime JS
+
+<button @onclick="async () => {ToggleTheme();}" class="btn btn-outline-@GetButtonStyle"><i class="fa-solid fa-@GetThemeIcon"></i></button>
+
+@code {
+    bool prefersLightTheme = true;
+
+    string GetTheme => prefersLightTheme ? "light" : "dark";
+
+    string GetButtonStyle => prefersLightTheme ? "dark" : "light";
+
+    string GetThemeIcon => prefersLightTheme ? "sun" : "moon";
+
+    protected override async Task OnAfterRenderAsync(bool firstRender)
+    {
+        if (!firstRender) return;
+
+        var themePref = await JS.InvokeAsync<string>("getThemePreference", new object { });
+
+        if (themePref == "dark")
+        {
+            prefersLightTheme = false;
+        }
+        else
+        {
+            prefersLightTheme = true;
+        }
+
+    }
+
+    async void ToggleTheme()
+    {
+        prefersLightTheme = !prefersLightTheme;
+        await JS.InvokeVoidAsync("saveThemePreference", GetTheme);
+        await JS.InvokeVoidAsync("applyTheme", GetTheme);
+        StateHasChanged();
+    }
+}
```

Comment:
Instead of a toggle I feel it would be better to support `SetThemeTo(Theme)`. Toggle only locks us into 2 themes without major refactoring, and it would be better to do that refactoring now.

Reply:
This is for managing the bootstrap color scheme which only supports dark and light mode.